java/flight/flight-core/src/main/java/org/apache/arrow/flight/grpc/AddWritableBuffer.java

```diff
@@ -88,11 +87,8 @@
    * @param buf The buffer to add.
    * @param stream The Candidate OutputStream to add to.
    * @return True if added. False if not possible.
-   * @throws IOException on error
    */
-  public static boolean add(ByteBuf buf, OutputStream stream) throws IOException {
-    buf.readBytes(stream, buf.readableBytes());
```

Comment:
shouldn't this now be wrapped in FAST_PATH?

Reply:
Ah, FAST_PATH is an existing flag - it's actually used at the callsite: https://github.com/apache/arrow/blob/6b85f6eba6d8403dbd549212fbc86f0e8ee192d0/java/flight/flight-core/src/main/java/org/apache/arrow/flight/ArrowMessage.java#L437-L439

But I agree it'd be clearer if we moved FAST_PATH into here and additionally had it handle the fallback path.

Reply:
Moved the fallback path into this method (mimicking GetReadableBuffer).